### DISCUSSION TITLE
Problems opening multiple windows

### DISCUSSION THREAD

**Discussion by vocatart** [AvaloniaUI/Avalonia#6209] [Q&A] [answered]
Hello, I'm a bit new to Avalonia, so I apologize if the answer to this question is obvious, but I looked around for a while, and looked at a few other apps and still can't figure out how to do this.

Simply put, I currently have a task that opens a file dialog, gets a file, and then opens two windows.

        public async Task ProjectFileDialog()
        {
            var result = await ShowOpenFileDialog.Handle(Unit.Default);

            if (result is object)
            {
                // Get target file
                // This file will be deserialized in WaveformWindow and ParameterWindow
                TargetFile = result;

                // Open waveform and parameter windows
                var waveformWindow = new configyr.Views.WaveformWindow();
                var parameterWindow = new configyr.Views.ParameterWindow();

                parameterWindow.Show();
                waveformWindow.Show();
            }
        }

The above method works and opens the two windows after the file is chosen, but the problem I am experiencing is that the two opened windows are not able to find their respective ViewModels, and thus aren't able to display data.

From what I can assume it's because the DataContext isn't set properly, but I have no idea how to go about ensuring that the windows are able to have the correct DataContext.

Thank you :)

**Comment by maxkatz6** [comment, marked as answer]
Similarly as in other XAML frameworks, you need to set DataContext by yourself.
Create object of view model with parameters you want (or resolve from IoC container), and assign to window.DataContext before or after showing it.

**Reply by vocatart** [reply]
perfect! this was just what I needed! Thanks !! :)
